modules/renter/repair.go

```diff
@@ -150,6 +151,21 @@ func (f *file) chunkHosts(chunk uint64) []modules.NetAddress {
 	return old
 }
 
+// expiringContracts returns the contracts that will expire soon.
+// TODO: what if contract has fully expired?
+func (f *file) expiringContracts(height types.BlockHeight) []fileContract {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	var expiring []fileContract
+	for _, fc := range f.contracts {
+		if height > fc.WindowStart-renewThreshold {
```

Comment:
so, if you are calling the `expiringContracts` function, at what stage are the hosts pruned that are either offline or whose contracts have fully expired?

I suppose they could be managed by the hostdb, if the host is offline or the contract has fully expired, the hostdb just skips over the request (maybe returning an error?)